### ISSUE TITLE
Implement `IN` keyword for local evaluation

### ISSUE THREAD

**Issue by khvn26** [Flagsmith/flagsmith-nodejs-client#124]
Part of https://github.com/Flagsmith/flagsmith/issues/1595.

NodeJS client's local evaluation mode has to support the `IN` keyword for segments. 

Refer to the Python engine implementation: https://github.com/Flagsmith/flagsmith-engine/blob/4cfb30792feb1fe9f64c082af7e9d64fc1a2f451/flag_engine/segments/models.py#L75-L80

# Acceptance Criteria

- [ ] When initialized with local evaluation, Flagsmith client correctly evaluates segments defined with the `IN` keyword.

**Comment by novakzaballa** [comment]
Tested by creating a segment called "Colors", and adding a condition using the `IN` keyword as shown below:


![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/26ac4b07-45f0-40ab-b922-5b80c409cfb4)

In `ff_segment` flag the value is "segment" and was created a segment override with the "Colors" segment

![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/8110838a-f673-4f67-9a03-304e9333a60d)

I used an identity named "user1" and added a new trait `{color: "red"}`. Also, the value of `ff_segment` in its segments was changed to "value".

![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/154985ce-7637-4de4-9229-256f15c8cb99)


Created an application in Node.js to test the result of `getIdentityFlags` and check if it is correct, the value obtained when using the `getEnvironmentFlags` method was also tested.

The results are shown below the following image.


![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/ee94a3e8-2825-4440-a5b6-2d32ac475bac)

Here you can see that the value of  `ff_segment` is "value"

![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/59074fdd-052c-4d37-a799-e7c5aa788aba)

Here you can see that the value of  `ff_segment` is "segment"

![Image](https://github.com/Flagsmith/flagsmith-nodejs-client/assets/41410593/22e21cd7-1cf6-42a8-8c36-5e0870d9ed15)

The results are correct in both cases.

In conclusion, the IN keyword seems to be working as expected, and no need for changes was detected.
